addons/web/static/tests/core/network/http_service.test.js

```diff
@@ -0,0 +1,44 @@
+/** @odoo-module **/
+
+import { after, expect, test } from "@odoo/hoot";
+import { mockFetch } from "@odoo/hoot-mock";
+import { get, post } from "@web/core/network/http_service";
+
+test`headless`("method is correctly set", async () => {
+    const restoreFetch = mockFetch((_, { method }) => {
+        expect.step(method);
+    });
+    after(restoreFetch);
+
+    await get("/call_get");
+    expect(["GET"]).toVerifySteps();
+
+    await post("/call_post");
+    expect(["POST"]).toVerifySteps();
+});
+
+test`headless`("check status 502", async () => {
+    const restoreFetch = mockFetch(() => new Response("{}", { status: 502 }));
+    after(restoreFetch);
+    await expect(() => get("/custom_route")).rejects.toThrow();
```

Comment:
no specific error to expect here?

Reply:
yes and I've noticed this assertion passed thanks to an error thrown by hoot while adding the matcher.